bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/ViewsPreferencePage.java

```diff
@@ -180,6 +190,50 @@ private void createThemeIndependentComposits(Composite comp) {
 		createEnableMruPref(comp);
 	}
 
+	protected void createShowFullText(Composite composite) {
+		IEclipsePreferences prefs = getSwtRendererPreferences();
+		boolean actualValue = prefs.getBoolean(CTabRendering.SHOW_FULL_TEXT, CTabRendering.SHOW_FULL_TEXT_DEFAULT);
+		createLabel(composite, ""); //$NON-NLS-1$
+		createLabel(composite, WorkbenchMessages.ViewsPreference_viewTabs_icons_and_titles_label);
+		showFullText = createCheckButton(composite, WorkbenchMessages.ViewsPreference_showFullTabText,
+				actualValue);
+	}
+
+	protected void createHideIcons(Composite composite) {
+		IEclipsePreferences prefs = getSwtRendererPreferences();
+		boolean actualValue = prefs.getBoolean(CTabRendering.HIDE_ICONS, CTabRendering.HIDE_ICONS_DEFAULT);
+		hideIcons = createCheckButton(composite, WorkbenchMessages.ViewsPreference_hideTabIcons, actualValue);
+	}
+
+	/**
+	 * @param showFullText
+	 * @param hideIcons
+	 */
+	private void createDependency(Button parent, Button dependent) {
+		GridData gridData = new GridData();
+		gridData.horizontalIndent = 20;
+		dependent.setLayoutData(gridData);
+
+		boolean parentState = parent.getSelection();
+		dependent.setEnabled(parentState);
+
+		SelectionListener listener = new SelectionListener() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				boolean state = parent.getSelection();
+				dependent.setEnabled(state);
+
+				if (state == false)
```

Comment:
Formatting seems not to be according to the standards. Didn't you format it? Shouldn't it be 
if (!state) {
...